### PR TITLE
Fix 1958 Filter wellbore object types

### LINKED ISSUE #1958: Filter wellbore object types
Implement object filter to hide object group menu items from the sidebar and the content view. A stub of the GUI for picking which objects to show is already implemented as shown in the screenshot below. Make a GetCap call when connecting to a WITSML server to check whether any unsupported object types should be permanently disabled.

Solves https://github.com/equinor/witsml-explorer/issues/662

image

### PR DESCRIPTION
**Pull request by eliasbruvik**
[//]: # (Request Template Witsml Explorer)

[//]: # (Thank you for contributing to WITSML Explorer! Before submitting this PR, please fill in the following template.)

## Fixes

[//]: # (Write the GitHub issue number starting with #, or, for Equinor only, the Jira issue number starting with WE-)

This pull request fixes #1958 

## Description

[//]: # (Please include a written summary of the changes and which issue is fixed. List any dependencies that are required for this change._)

* Added ServerCapabilities models.
* Added a route in the API to get the current server's capabilities.
* A bit cleaning of the FilterPanel component.
* Added objectVisibilityStatus to Filter to mark an object as visible, hidden or disabled.
* Fetch the capabilities in ServerManager after connecting to a new server.
* Objects supported by WEx that are not in the server capabilities are set as disabled.
* Only objects marked as visible are shown in the navbar and WellboreObjectTypesListView.

## Type of change

[//]: # (Mark any of the types of change that apply.)

* [ ] Bugfix
* [x] New feature (non-breaking change which adds functionality)
* [x] Enhancement of existing functionality
* [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update

## Impacted Areas in Application

[//]: # (List general components of the application that this PR will affect)

* [x] Frontend
* [x] API
* [x] WITSML
* [ ] Other (please describe)

## Checklist:

[//]: # (Please tick all the boxes or remove the ones that aren't needed and explain why)

*Communication*
* [ ] I have made corresponding changes to the documentation
* [ ] PR affects application security

*Code quality*
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [ ] New code is covered by passing tests

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: Src/Witsml/Data/WitsmlServerCapabilities.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    public class WitsmlFunctionDataObject
+    {
+        [XmlText]
+        public string Name { get; init; }
+
+        [XmlAttribute("maxDataNodes")]
+        public int MaxDataNodes { get; init; }
+
+        [XmlElement("maxDataPoints")]
```
Comment (janburak) [inline Src/Witsml/Data/WitsmlServerCapabilities.cs]: Should be XmlAttribute

File: Src/Witsml/Data/WitsmlClientCapabilities.cs [hunk 2]
```diff
@@ -44,7 +44,7 @@ public string ToXml()
         }
     }
 
-    public class ClientContactInformation
+    public class ContactInformation
```
Comment (janburak) [inline Src/Witsml/Data/WitsmlClientCapabilities.cs:47]: I think that renaming classes in the Witsml project counts as a "breaking change", due to other projects that use the Witsml nuget package having to rename the class on their end.

File: Src/WitsmlExplorer.Frontend/components/ContentViews/ServerManager.tsx [hunk 3]
```diff
@@ -49,7 +53,12 @@ const ServerManager = (): React.ReactElement => {
         return;
       }
       try {
-        const wells = await WellService.getWells();
+        const [wells, supportedObjects] = await Promise.all([WellService.getWells(), CapService.getCapObjects()]);
+        const updatedVisibility = { ...selectedFilter.objectVisibilityStatus };
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/components/ContentViews/ServerManager.tsx]: When connecting to a server that does not support an object, and then connecting to another server that does support that object, the object will remain disabled due to reusing `selectedFilter.objectVisibilityStatus` and the following code only disabling objects, but not enabling them back.

File: Src/Witsml/Data/WitsmlServerCapabilities.cs (hunk 1 above)
Comment (eliasbruvik) [reply Src/Witsml/Data/WitsmlServerCapabilities.cs]: Right, should be fixed now.

File: Src/Witsml/Data/WitsmlClientCapabilities.cs (hunk 2 above)
Comment (eliasbruvik) [reply Src/Witsml/Data/WitsmlClientCapabilities.cs:47]: PR marked as "breaking change".

File: Src/WitsmlExplorer.Frontend/components/ContentViews/ServerManager.tsx (hunk 3 above)
Comment (eliasbruvik) [reply Src/WitsmlExplorer.Frontend/components/ContentViews/ServerManager.tsx]: Should be fixed now.

**Review by janburak — APPROVED** [review]
- [X] Read issue and PR
- [X] Pulled branch and manually tested
- [X] Verify that PR resolves issue
- [X] Reviewed the code

**Pull request merged**